src/language_negotiation_lambda.js

```diff
@@ -24,32 +37,97 @@ exports.handler = (event, context, callback) => {
                 sanitizedWeight = parseFloat(weight.substring(2)) || 0.0;
             }
 
-            if (supportedLanguages.includes(sanitizedLocale)){
-                selectableLanguages.push({
-                    'locale': sanitizedLocale,
-                    'weight': sanitizedWeight
-                });
+            sanitizedLanguages.push({
+                'locale': sanitizedLocale,
+                'weight': sanitizedWeight
+            });
+        });
+
+        return sanitizedLanguages;
+    }
+
+    const getSelectableLanguages = function(languages){
+        var selectableLanguages = [];
+        languages.forEach(function(language){
+            if (supportedLanguages.indexOf(language.locale) >= 0){
+                selectableLanguages.push(language);
             }
         });
 
         return selectableLanguages;
     }
 
-    try {
-        if (headers[acceptLanguage]) {
-            var selectableLanguages = getSelectableLangugaes(headers[acceptLanguage].split(','));
-            var selectedLanguage = defaultLanguage;
-            if(selectableLanguages.length > 0) {
+    const getCookieLanguage = function(cookies) {
+        var languageCookieName,
+            languageCookieValue,
+            locales,
+            selectableLanguages,
+            regexp;
+
+        try{
+            languageCookieName = process.env.LANGUAGE_COOKIE_NAME;
+            // Regex from https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie#Example_2_Get_a_sample_cookie_named_test2
+            regexp = new RegExp("(?:(?:^|.*;\\s*)" + languageCookieName + "\\s*\\=\\s*([^;]*).*$)|^.*$");
+            locales = [];
+            if(cookies){
+                cookies.forEach(function(cookies_str){
+                    languageCookieValue = cookies_str.replace(regexp, "$1");
+                    locales.push({ locale: sanitizeLocale(languageCookieValue) });
+                });
+            }
+
+            return getLocale(getSelectableLanguages(locales));
+        } catch(err){
+            console.log("Error fetching cookie data: " + err);
+        }
+    }
+
+    const getBrowserLanguage = function(language_headers) {
+        var selectableLanguages = [];
+
+        try{
+            if (language_headers) {
+                language_headers.forEach(function(language_str){
+                    var languages = parseAcceptLanguage(language_str);
+                    var selectable = getSelectableLanguages(languages);
+                    selectableLanguages = selectableLanguages.concat(selectable);
+                });
+
                 selectableLanguages.sort(function(a, b){
                     return a.weight - b.weight; // Sort Ascending
                 });
```

Comment:
Does the order of the languages in the header have any bearing on weight? What typically happens when languages in the header all have the same weight?